main-command/src/main/java/sbt/internal/BootServerSocket.java

```diff
@@ -337,18 +336,24 @@ public void close() {
   static ServerSocket newSocket(final String sock) throws ServerAlreadyBootingException {
     ServerSocket socket = null;
     String name = socketName(sock);
+    boolean jni = requiresJNI() || System.getProperty("sbt.ipcsocket.jni", "false").equals("true");
```

Comment:
I feel that this `requiresJNI` should be implemented inside `ipcsocket`.

`sbt` is client for `ipcsocket` and `ipcsocket` may include or may not include some library.

Reply:
Maybe, but I don't think it's necessary. The ipcsocket is agnostic on the implementation and provides constructors that take a `useJNI` flag so downstream consumers can freely choose their implementation and copy this logic if they feel they need it.

Reply:
@eatkins right now it looks a bit strange. Library A doesn't support feature because it has limitation and all this limitation is implementing on client of this library.

From my point of view this check should be implemented inside, and API should have "force JNI" that end user can control.

Reply:
I don't know. You may be right but I'm not sure what the best approach is and having `requiresJNI` in sbt was easier to implement and has no downstream effect. All downstream consumers are likely using the jna implementation because it was a very recent change to support jni at all. You're welcome to open a pr in sbt ipcsocket but I'm not motivated to change this code or add it there.